plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+@file:Suppress("UnstableApiUsage")
+
+package io.sentry.android.gradle.telemetry
+
+import io.sentry.BuildConfig
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.ITransaction
+import io.sentry.NoOpHub
+import io.sentry.Sentry
+import io.sentry.SentryEvent
+import io.sentry.SentryLevel
+import io.sentry.SpanStatus
+import io.sentry.android.gradle.SentryPlugin
+import io.sentry.android.gradle.SentryPropertiesFileProvider
+import io.sentry.android.gradle.extensions.SentryPluginExtension
+import io.sentry.android.gradle.telemetry.SentryCliInfoValueSource.Params
+import io.sentry.android.gradle.util.AgpVersions
+import io.sentry.android.gradle.util.GradleVersions
+import io.sentry.android.gradle.util.SentryCliException
+import io.sentry.android.gradle.util.getBuildServiceName
+import io.sentry.exception.ExceptionMechanismException
+import io.sentry.gradle.common.SentryVariant
+import io.sentry.protocol.Mechanism
+import io.sentry.protocol.User
+import java.io.ByteArrayOutputStream
+import java.nio.charset.Charset
+import javax.inject.Inject
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.internal.tasks.execution.ExecuteTaskBuildOperationDetails
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ValueSource
+import org.gradle.api.provider.ValueSourceParameters
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters.None
+import org.gradle.api.tasks.Input
+import org.gradle.execution.RunRootBuildWorkBuildOperationType
+import org.gradle.internal.operations.BuildOperationDescriptor
+import org.gradle.internal.operations.BuildOperationListener
+import org.gradle.internal.operations.OperationFinishEvent
+import org.gradle.internal.operations.OperationIdentifier
+import org.gradle.internal.operations.OperationProgressEvent
+import org.gradle.internal.operations.OperationStartEvent
+import org.gradle.process.ExecOperations
+import org.gradle.util.GradleVersion
+
+abstract class SentryTelemetryService :
+    BuildService<None>,
+    BuildOperationListener,
+    AutoCloseable {
+
+    private var hub: IHub = NoOpHub.getInstance()
+    private var transaction: ITransaction? = null
+    private var didAddChildSpans: Boolean = false
+    private var started: Boolean = false
+
+    fun start(startParameters: SentryTelemetryServiceParams) {
+        if (started) {
+            return
+        }
+
+        started = true
+        try {
+            if (startParameters.saas == false) {
+                SentryPlugin.logger.info(
+                    "Sentry is running against a self hosted instance. Telemetry has been disabled."
+                )
```

Comment:
```suggestion
                SentryPlugin.logger.info {
                    "Sentry is running against a self hosted instance. Telemetry has been disabled."
                }
```
this will prepend the message with `[sentry]`

Reply:
I'd also replace it with our logger extension for other logs down here